### PR TITLE
Adblocking test with ad-free offering. 

### PR DESCRIPTION
**Pull request by JustinPinner**
## What does this change?

This is the second iteration of the adblock engagement test (https://github.com/guardian/frontend/pull/14072). We're exploring the appetite for an ad-free offering, targeted initially at our ad-blocking readers.

**This is a ZERO PERCENT test** for validating everything prior to deploying the real deal.
## What is the value of this and can you measure success?

We're exploring the possibility that a significant proportion of ad-blocking readers are interested in paying for an ad-free experience.
## Does this affect other platforms - Amp, Apps, etc?

No.

## Screenshots

screen shot 2016-09-27 at 14 37 28

screen shot 2016-09-27 at 14 37 54

screen shot 2016-09-27 at 14 38 12
## Request for comment

@regiskuckaertz @rich-nguyen @kenlim

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js [hunk 1]
```diff
@@ -17,75 +15,91 @@ define([
...
-                    $('.survey-adblock').addClass('is-hidden');
-                    var cookieName = 'gu_abm_x',
-                        cookieLifetimeMinutes = 30,
-                        cookieCount = cookies.get(cookieName) ? parseInt(cookies.get(cookieName)) : 0;
-                    cookies.addForMinutes(cookieName, cookieCount + 1, cookieLifetimeMinutes);
-                });
-            }
+            bean.on(document, 'click', $('.cta-whitelist'), function () {
+                // -> state 2
+                $.forEachElement(('.state-1'), function(el){el.classList.add('is-hidden');});
```
Comment (dominickendrick) [inline static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js]: Can all these `DOM` changes be wrapped in a `fastdom.write` ?

Comment (JustinPinner) [inline static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js]: Doh. They should have been :(

File: static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js [hunk 2]
```diff
@@ -17,75 +15,91 @@ define([
...
+                $.forEachElement(('.state-3'), function(el){el.classList.add('is-hidden');});
+                $.forEachElement(('.state-4'), function(el){el.classList.add('is-hidden');});
+                $.forEachElement(('.state-2'), function(el){el.classList.remove('is-hidden');});
+                $('.survey-container').removeClass('thank-you');
+            });
+            bean.on(document, 'click', $('.survey-button-rounded__cta.noads'), function () {
+                // -> state 3
+                $.forEachElement(('.state-1'), function(el){el.classList.add('is-hidden');});
```
Comment (dominickendrick) [inline static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js]: Could you abstract out these show/hide functions, there seems to be a lot of duplication

File: static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js (hunk 1 above)
Comment (JustinPinner) [inline static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js]: Hold on, they are wrapped in a fastdom.write. Does that count?

File: static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js [hunk 3]
```diff
@@ -17,75 +15,91 @@ define([
...
+                $.forEachElement(('.state-2'), function(el){el.classList.add('is-hidden');});
+                $.forEachElement(('.state-4'), function(el){el.classList.add('is-hidden');});
+                $.forEachElement(('.state-3'), function(el){el.classList.remove('is-hidden');});
+                $('.survey-container').removeClass('thank-you');
+            });
+            bean.on(document, 'click', $('.survey-button-rounded__cta.paypal'), function () {
+                // -> state 4
+                $.forEachElement(('.state-1'), function(el){el.classList.add('is-hidden');});
```
Comment (dominickendrick) [inline static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js]: Can these `state-x` be given more descriptive names ?

File: static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js (hunk 1 above)
Comment (dominickendrick) [inline static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js]: oh yeah ;) my bad !

File: static/src/javascripts/projects/common/views/commercial/survey/survey-adblock.html [hunk 4]
```diff
@@ -1,87 +1,113 @@
...
+        </div>
+
+        <div class="survey-buttons">
+            <div class="survey-button-rounded__cta noads state-1 state-2" data-link-name="<%=adFreeDataLink%>"><%=adFreeButtonText%><%=arrowWhiteRight%></div>
+            <div class="survey-button-rounded__cta cta-whitelist state-1" data-link-name="<%=whitelistDataLink%>"><%=whitelistText%><%=arrowWhiteRight%></div>
+            <div class="survey-button-rounded__cta readon state-4 is-hidden" data-link-name="adblock paymentdone continuetoarticle">Continue to article<%=arrowWhiteRight%></div>
+        </div>
+
+        <div class="howto-unblock state-2 is-hidden">
+            <hr />
```
Comment (dominickendrick) [inline static/src/javascripts/projects/common/views/commercial/survey/survey-adblock.html]: I'm not quite sure this is a semantic use of the `<hr />` element:

[MDN](https://developer.mozilla.org/en/docs/Web/HTML/Element/hr) says:
> The HTML `<hr>` element represents a thematic break between paragraph-level elements (for example, a change of scene in a story, or a shift of topic with a section)

If its just for styling, use `border-top` etc

File: static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js (hunk 3 above)
Comment (JustinPinner) [inline static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js]: Done.

File: static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js (hunk 2 above)
Comment (JustinPinner) [inline static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js]: Done.

File: static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js (hunk 1 above)
Comment (JustinPinner) [inline static/src/javascripts/projects/common/modules/commercial/survey/survey-adblock.js]: 😄

File: static/src/javascripts/projects/common/views/commercial/survey/survey-adblock.html (hunk 4 above)
Comment (JustinPinner) [inline static/src/javascripts/projects/common/views/commercial/survey/survey-adblock.html]: > In previous versions of HTML, it represented a horizontal rule.

Sigh.

Comment (JustinPinner) [inline static/src/javascripts/projects/common/views/commercial/survey/survey-adblock.html]: Done.

File: static/src/javascripts/projects/common/modules/experiments/tests/adblocking-response.js [hunk 5]
```diff
@@ -54,20 +52,18 @@ define([
         };
 
         var isQualified = function () {
```
Comment (dominickendrick) [inline static/src/javascripts/projects/common/modules/experiments/tests/adblocking-response.js]: This should probably be in `canRun` but you seem to have complex requirements, this should be discussed with @desbo before we put the actual test live

**Comment by dominickendrick** [comment]
👍

**Pull request merged**